release/models/system/openconfig-aaa.yang

```diff
@@ -194,6 +266,49 @@ module openconfig-aaa {
     }
   }
 
+  grouping aaa-server-config-data {
+    description
+      "Common configuration data for AAA servers";
+
+    leaf name {
+      type string;
+      description
+        "Name assigned to the server";
+    }
+
+    leaf address {
+      type oc-inet:ip-address;
+      description "Address of the authentication server";
+    }
+
+    leaf port {
+      type oc-inet:port-number;
+      description "Port of the authentication server";
+    }
+
+    leaf network-instance {
+      type oc-ni:network-instance-ref;
+      description "Network-instance of the authentication server";
+    }
+
+    leaf type {
```

Comment:
This aaa server type overlaps/conflicts with the type defined at the server-group level.  Do we need this defined per server?

Also, I observe that there is only one base type, so maybe we don't really even need type at all?

Reply:
Thanks for reviewing Darren.
`This aaa server type overlaps/conflicts with the type defined at the server-group level. Do we need this defined per server?`
An AAA server can be used either directly by itself or within a server group, so we need this type defined per server as well. Vendors can decide what to do if they see a conflict between the type of the group and the type of the server.

`Also, I observe that there is only one base type, so maybe we don't really even need type at all?`
`leaf type` here is an identityref which has existing identity RADIUS and TACACS added to it from their own files:
https://github.com/openconfig/public/blob/1d1edd12ad5e8dc191b1d01fe53d55f72bf893d8/release/models/system/openconfig-aaa-radius.yang#L74
https://github.com/openconfig/public/blob/1d1edd12ad5e8dc191b1d01fe53d55f72bf893d8/release/models/system/openconfig-aaa-tacacs.yang#L73
so it seems to have two types with it. Am I using this wrong?